src/test/scala/is/hail/methods/SkatSuite.scala

```diff
@@ -0,0 +1,450 @@
+        (keysType, filteredVds.rdd.flatMap { case (v, (va, gs)) =>
+          (Option(keysQuerier(va)), Option(weightQuerier(va))) match {
+            case (Some(key), Some(w)) =>
+              val x = getGenotype(gs, n)
+              Some((key, (x, w.asInstanceOf[Double])))
+            case _ => None
+          }
+        })
+      } else {
+        val keyType = keysType match {
+          case TArray(e) => e
+          case TSet(e) => e
+          case _ => fatal(s"With single_key=False, variant keys must be of type Set[T] or Array[T], got $keysType")
+        }
+        (keyType, filteredVds.rdd.flatMap { case (v, (va, gs)) =>
+          val keys = Option(keysQuerier(va).asInstanceOf[Iterable[_]]).getOrElse(Iterable.empty)
+          val optWeight = Option(weightQuerier(va))
+          if (keys.isEmpty || optWeight.isEmpty)
+            Iterable.empty
+          else {
+            val x = getGenotype(gs, n)
+            keys.map((_, (x, optWeight.get.asInstanceOf[Double])))
+          }
+        })
+      }
+
+    val aggregatedKT = keyedRdd.aggregateByKey(zero)(SkatAggForR.seqOp, SkatAggForR.combOp)
+
+    val inputFilePheno = tmpDir.createLocalTempFile("skatPhenoVec", ".txt")
+    hadoopConf.writeTextFile(inputFilePheno) {
+      _.write(largeMatrixToString(y.toDenseMatrix, " "))
+    }
+
+    val inputFileCov = tmpDir.createLocalTempFile("skatCovMatrix", ".txt")
+    hadoopConf.writeTextFile(inputFileCov) {
+      _.write(largeMatrixToString(cov, " "))
+    }
+
+    aggregatedKT.collect().map { case (key, safr) => {
+      val (xs, weights) = resultOp(safr, n)
+      
+      //write files to a location R script can read
+      val inputFileG = tmpDir.createLocalTempFile("skatGMatrix", ".txt")
+      hadoopConf.writeTextFile(inputFileG) {
+        _.write(largeMatrixToString(xs, " "))
+      }
+
+      val inputFileW = tmpDir.createLocalTempFile("skatWeightVec", ".txt")
+      hadoopConf.writeTextFile(inputFileW) {
+        _.write(largeMatrixToString(weights.toDenseMatrix, " "))
+      }
+
+      val resultsFile = tmpDir.createLocalTempFile("results", ".txt")
+
+      val rScript = s"Rscript src/test/resources/skatTest.R " +
+        s"${ uriPath(inputFileG) } ${ uriPath(inputFileCov) } " +
+        s"${ uriPath(inputFilePheno) } ${ uriPath(inputFileW) } " +
+        s"${ uriPath(resultsFile) }"
+
+      rScript !
+      val results = readResults(resultsFile)
+
+      Row(key, results(0)(0), results(0)(1))
+    }
+    }
+
+  }
+
+
+  def covariates = hc.importTable("src/test/resources/regressionLinear.cov",
+    types = Map("Cov1" -> TDouble, "Cov2" -> TDouble)).keyBy("Sample")
+
+  def phenotypes = hc.importTable("src/test/resources/regressionLinear.pheno",
+    types = Map("Pheno" -> TDouble), missing = "0").keyBy("Sample")
+
+  def intervals = IntervalList.read(hc, "src/test/resources/regressionLinear.interval_list")
+
+  def vdsBurden: VariantDataset = hc.importVCF("src/test/resources/regressionLinear.vcf")
```

Comment:
burden is typically used for the other kind of gene tests. here you could just call it vds unless that conflicts with elsewhere.